File: Python/1-25/10.py
#!/usr/bin/python
import math
n = 2000000; pl = [False] * 2 + [True] * (n-1)
for i in range(2, int(math.sqrt(n))+2):
    if pl[i]:
        for j in range(i**2, n+1, i): pl[j] = False
print(sum([i for i in range(len(pl)) if pl[i]]))

File: C/1-25/10.c
#include <stdio.h>
#include <math.h>
int main(int argc, char *argv[])
{
    unsigned long int i, j, limit = 2000000, sum=0;
    int bool_list[limit];
    for (i=2; i < limit; i++)
        bool_list[i] = 1;
    for (i=2; i < sqrt(limit); i++) 
        if (bool_list[i])
            for (j=i*i; j <= limit; j += i)
                bool_list[j] = 0;
    for (i=2; i < limit; i++)
        if (bool_list[i])
            sum += i;
    printf("%lu", sum);
    return 0;
}
